packages/terra-clinical-data-grid/src/subcomponents/SectionHeader.jsx

```diff
@@ -0,0 +1,136 @@
+import IconCaretDown from 'terra-icon/lib/icon/IconCaretDown';
+
+import { KEYCODES } from '../utils/keycodes';
+
+import styles from './SectionHeader.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String identifier of the section that the SectionHeader is rendered within.
+   */
+  sectionId: PropTypes.string.isRequired,
+  /**
+   * String text rendered within the SectionHeader's default text position.
+   */
+  text: PropTypes.string,
+  /**
+   * Content rendered within the SectionHeader's default leading content position (ahead of the text).
+   */
+  startAccessory: PropTypes.node,
+  /**
+   * Content rendered within the SectionHeader's default trailing content position (after the text).
+   */
+  endAccessory: PropTypes.node,
+  /**
+   * Boolean indicating whether the SectionHeader is representing a collapsible section. If true, the SectionHeader
+   * will be selectable, and an collapsiblity indicator will be rendered within the SectionHeader.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Boolean indicating whether the SectionHeader is representing a collapsed section.
+   */
+  isCollapsed: PropTypes.bool,
+  /**
+   * Function called upon selection of the SectionHeader. The `isCollapsible` prop must be true for this function
+   * to be executed. Parameters: `onRequestSectionCollapse(sectionId)`
+   */
+  onRequestSectionCollapse: PropTypes.func,
+  /**
+   * Function that will be called with a ref to the SectionHeader's selectable element. Parameters: `selectableRefCallback(selectableRef)`
+   */
+  selectableRefCallback: PropTypes.func,
+  /**
+   * Content to be rendered within the SectionHeader. If provided, the `text`, `startAccessory`, and `endAccessory` props are ignored.
+   */
+  children: PropTypes.node,
+};
+
+class SectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClick = this.handleClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+
+  handleClick() {
+    const { onRequestSectionCollapse, sectionId } = this.props;
+
+    if (onRequestSectionCollapse) {
+      onRequestSectionCollapse(sectionId);
+    }
+  }
+
+  handleKeyDown(event) {
+    if (event.nativeEvent.keyCode === KEYCODES.ENTER || event.nativeEvent.keyCode === KEYCODES.SPACE) {
+      const { onRequestSectionCollapse, sectionId } = this.props;
+
+      if (onRequestSectionCollapse) {
+        event.preventDefault();
+        onRequestSectionCollapse(sectionId);
+      }
+    }
+  }
+
+  render() {
+    const {
+      text, startAccessory, endAccessory, children, isCollapsible, isCollapsed, selectableRefCallback,
+    } = this.props;
+
+    let content;
+    if (children) {
+      content = (
+        <div className={cx('flex-content')}>
+          {children}
+        </div>
+      );
+    } else {
+      content = (
+        <React.Fragment>
+          <div className={cx('start-accessory')}>
```

Comment:
Do we need to check if start or end accessory are provided so we don't render extra divs?

Reply:
That would be great if we can do that. I have noticed that screen readers have been slightly off on what is read due to extra dom containers.

Reply:
Good idea, I can make that happen.